### ISSUE TITLE
Question about beam search.

### ISSUE THREAD

**Issue by smallv0221** [bytedance/lightseq#169]
Here is the code in step 3 of `select_beam_rough_topk()`.
![image](https://user-images.githubusercontent.com/33639025/131435734-30c4799d-cab9-434f-874c-705977b8ee3a.png)
The `idx` equals to left_idx. Which equals to `block_start + threadIdx.x`. Then it can not cover all the index in `vocab_size`. And the variable `iter` in this loop is not used in loop body. How could this loop select all the candidates token with logits bigger than `s_topk`.

**Comment by Taka152** [comment]
`idx` will change at each loop's end。
https://github.com/bytedance/lightseq/blob/a5844a287f08f3190438b2bfbab15bf5c0eb525d/lightseq/inference/kernels/transformerKernels.cc.cu#L160
